### PR TITLE
Fixed pom.xml

### PR DESCRIPTION
**Pull request by RedSmarty**
pom.xml was previously broken, it would not compile. This PR fixes those issues by adding version tag to maven jar plugin and updating lombok to 1.18.20. This also sets the target version as java 16. Set the new version in pom as 2.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -166,6 +167,15 @@
                     <dependencyReducedPomLocation>${project.build.directory}/dependency-reduced-pom.xml</dependencyReducedPomLocation>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (CoreyShupe) [inline pom.xml:172]: Why are you defining the compiler plugin &  the property compiler specs. The `<maven.compiler.source>16</maven.compiler.source>` takes care of this interaction already.

**Review by CoreyShupe — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (RedSmarty) [reply pom.xml]: I added that as a temporary check to see if it compiles properly on java 16. After bumping the lombok verison, it compiles fine. This has been removed.

File: pom.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
     <groupId>org.geysermc</groupId>
     <artifactId>packconverter</artifactId>
-    <version>1.0-SNAPSHOT</version>
+    <version>2.0</version>
```
Comment (rtm516) [inline pom.xml:35]:
```suggestion
    <version>2.0-SNAPSHOT</version>
```

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (RedSmarty) [reply pom.xml]: Bumped to 2.0-SNAPSHOT

**Review by rtm516 — APPROVED** [review]

**Pull request merged**